ethereum/scripts/dumpTestConfig.js

```diff
@@ -1,68 +1,60 @@
 const TOML = require('@iarna/toml');
 const fs  = require('fs');
-const os = require('os');
 const path = require('path');
 
-const Bridge = artifacts.require("Bridge")
-const ETHApp = artifacts.require("ETHApp")
-const ERC20App = artifacts.require("ERC20App")
-const TestToken = artifacts.require("TestToken")
-
-const jsonify = (abi) => JSON.stringify(abi, null, 2)
-
-const dumpConfig = async (tmpDir, contracts) => {
-
-    // Relayer Config
-    let bridgeAbiFile = path.join(tmpDir, "Bridge.json")
-    let ethAbiFile = path.join(tmpDir, "ETHApp.json")
-    let erc20AbiFile = path.join(tmpDir, "ERC20App.json")
+const channelContracts = {
+    basic: {
+        inbound: artifacts.require("BasicInboundChannel"),
+        outbound: artifacts.require("BasicOutboundChannel")
+    },
+    incentivized: {
+        inbound: artifacts.require("IncentivizedInboundChannel"),
+        outbound: artifacts.require("IncentivizedOutboundChannel")
+    },
+}
 
-    fs.writeFileSync(bridgeAbiFile, jsonify(contracts.bridge.abi))
-    fs.writeFileSync(ethAbiFile, jsonify(contracts.ethApp.abi))
-    fs.writeFileSync(erc20AbiFile, jsonify(contracts.erc20App.abi))
+const channels = {
+    basic: {
+        inbound: null,
+        outbound: null
+    },
+    incentivized: {
+        inbound: null,
+        outbound: null
+    },
+}
 
+const dump = (tmpDir, channels) => {
     const config = {
         ethereum: {
             endpoint: "ws://localhost:8545/",
             "descendants-until-final": 0,
-            bridge: {
-                address: contracts.bridge.address,
-                abi: bridgeAbiFile,
-            },
-            apps: {
-                eth: {
-                    address: contracts.ethApp.address,
-                    abi: ethAbiFile,
+            channels: {
+                basic: {
+                    inbound: channels.basic.inbound.address,
+                    outbound: channels.basic.outbound.address,
+                },
+                incentivized: {
+                    inbound: channels.incentivized.inbound.address,
+                    outbound: channels.incentivized.outbound.address,
                 },
-                erc20: {
```

Comment:
need to dump apps too

Reply:
Isn't necessary tho - `dumpTestConfig.js` actually only generates configuration for the relayer, and the relayer only needs to know the addresses for channels (by design its unaware of application contracts).

Since we use a ganache instance that generates deterministic addresses based upon a static seed, the E2E mocha tests and parachain genesis config have hardcoded addresses for the ETH and ERC20 apps. Is a very useful simplification, though the catch is if we change the order in which contracts are deployed then we would need to update the addresses everywhere.

Reply:
ah, yeh i thought the parachain genesis config used the dump.

on ropsten ofc the genesis will need to be changed to use whatever ropsten addresses we get. do we have a dump for this yet or do we just have to manually change the genesis config on ropsten?

Reply:
I've explained the process here a bit more in https://github.com/Snowfork/polkadot-ethereum/pull/237#discussion_r568577491